nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java

```diff
@@ -131,6 +133,7 @@
     @WritesAttribute(attribute = "merge.bin.age", description = "The age of the bin, in milliseconds, when it was merged and output. Effectively "
         + "this is the greatest amount of time that any FlowFile in this bundle remained waiting in this processor before it was output") })
 @SeeAlso({SegmentContent.class, MergeRecord.class})
+@SystemResourceConsideration(resource = SystemResource.MEMORY)
```

Comment:
It would probably be helpful here to add a description that explains that the content itself is not stored in memory but rather the FlowFiles' attributes and that the configuration for max bin size, etc. will influence how much heap is used. Would also call out that if merging together many small FlowFiles, a two-stage approach may be necessary in order to avoid running out of memory.